core/finality-grandpa/src/lib.rs

```diff
@@ -457,6 +458,7 @@ pub fn run_grandpa<B, E, Block: BlockT<Hash=H256>, N, RA>(
 	network: N,
 	inherent_data_providers: InherentDataProviders,
 	on_exit: impl Future<Item=(),Error=()> + Send + 'static,
+	telemetry_notify: Option<TelemetryHookOnConnect>,
```

Comment:
there are already a ton of parameters to this function already. can we refactor in a `Params` struct?

Reply:
Done :-).